### PR TITLE
feat: ELECTRON-1331 (Add new performance settings and add support for BI APIs)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Add new performance settings
Add support for BI instrumentation

[ELECTRON-1331](https://perzoinc.atlassian.net/browse/ELECTRON-1331)

### CODE REVIEW

File: src/app/analytic-events-handler.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ENABLED = 'ON',
+    DISABLED = 'OFF',
+}
+
+export enum AnalyticElements {
+    MENU = 'MENU',
+}
+
+const MAX_EVENT_QUEUE_LENGTH = 50;
+const analyticCallback = 'analytic-callback';
```
Comment (keerthi16) [inline src/app/analytic-events-handler.ts]: Typo: `analytics`

File: demo/index.html [hunk 2]
```diff
@@ -151,6 +153,7 @@ <h1>Symphony Electron API Demo</h1>
                 activate: 'activate',
                 registerBoundsChange: 'register-bounds-change',
                 registerProtocolHandler: 'register-protocol-handler',
+                registerAnalyticHandler: 'register-analytic-handler',
```
Comment (VishwasShashidhar) [inline demo/index.html]: please refactor this to `registerAnalyticsHandler`: `register-analytics-handler`

File: demo/index.html [hunk 3]
```diff
@@ -312,34 +315,30 @@ <h1>Symphony Electron API Demo</h1>
...
         var getVersionInfo = document.getElementById('get-version');
         getVersionInfo.addEventListener('click', () => {
+            let apiVersionInfo = document.getElementById('api-version');
+            let containerIdentifier = document.getElementById('container-identifier');
+            let version = document.getElementById('container-ver');
+            let buildNumber = document.getElementById('build-number');
+            let searchApiVer = document.getElementById('search-api-ver');
+            let cpuArch = document.getElementById('cpu-arch');
             if (window.ssf) {
                 ssf.getVersionInfo().then((verInfo) => {
```
Comment (VishwasShashidhar) [inline demo/index.html]: please refactor `verInfo` to `versionInfo`

File: demo/index.html [hunk 4]
```diff
@@ -436,6 +435,19 @@ <h1>Symphony Electron API Demo</h1>
             }
         });
 
+        /**
+         * Analytic events
+         */
+        const onAnalyticEvent = (data) => {
```
Comment (VishwasShashidhar) [inline demo/index.html]: refactor to `onAnalyticsEvent`

File: demo/index.html [hunk 5]
```diff
@@ -436,6 +435,19 @@ <h1>Symphony Electron API Demo</h1>
...
+            console.log(`event logger`, data);
+        };
+
+        if (window.ssf) {
+            window.ssf.registerAnalyticEvent(onAnalyticEvent);
```
Comment (VishwasShashidhar) [inline demo/index.html]: refactor to `registerAnalyticsEvent`

File: src/app/analytic-events-handler.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
+export interface IAnalyticData {
```
Comment (VishwasShashidhar) [inline src/app/analytic-events-handler.ts]: refactor the filename to `analytics-handler`, no need to have `events` in the middle.

File: src/app/analytic-events-handler.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+export enum MenuActionTypes {
+    AUTO_LAUNCH_ON_START_UP = 'auto_launch_on_start_up',
+    ALWAYS_ON_TOP = 'always_on_top',
+    MINIMIZE_ON_CLOSE = 'minimize_on_close',
+    FLASH_NOTIFICATION_IN_TASK_BAR = 'flash_notification_in_task_bar',
+    HAMBURGER_MENU = 'hamburger_menu',
+    REFRESH_APP_IN_IDLE = 'refresh_app_in_idle',
+}
+
+export enum AnalyticActions {
```
Comment (VishwasShashidhar) [inline src/app/analytic-events-handler.ts]: refactor to `AnalyticsActions`

File: src/app/analytic-events-handler.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    ENABLED = 'ON',
+    DISABLED = 'OFF',
+}
+
+export enum AnalyticElements {
```
Comment (VishwasShashidhar) [inline src/app/analytic-events-handler.ts]: refactor to `AnalyticsElements`

File: src/app/analytic-events-handler.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    MENU = 'MENU',
+}
+
+const MAX_EVENT_QUEUE_LENGTH = 50;
+const analyticCallback = 'analytic-callback';
+
+class Analytics {
+    private preloadWindow: Electron.webContents | undefined;
+    private analyticEventQueue: IAnalyticData[] = [];
```
Comment (VishwasShashidhar) [inline src/app/analytic-events-handler.ts]: refactor to `analyticsEventQueue`

File: src/app/analytic-events-handler.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * Stores the reference to the preload window
+     *
+     * @param webContents {Electron.webContents}
+     */
+    public registerPreloadWindow(webContents: Electron.webContents): void {
+        this.preloadWindow = webContents;
+
+        if (this.preloadWindow && !this.preloadWindow.isDestroyed()) {
```
Comment (VishwasShashidhar) [inline src/app/analytic-events-handler.ts]: flip the if statements and return early, cleaner code that way.

File: src/app/app-menu.ts [hunk 11]
```diff
@@ -3,6 +3,12 @@ import { app, dialog, Menu, MenuItemConstructorOptions, session, shell } from 'e
 import { isMac, isWindowsOS } from '../common/env';
 import { i18n, LocaleType } from '../common/i18n';
 import { logger } from '../common/logger';
+import {
+    AnalyticActions,
```
Comment (VishwasShashidhar) [inline src/app/app-menu.ts]: refactor to `Analytics`

File: src/app/chrome-flags.ts [hunk 12]
```diff
@@ -1,12 +1,15 @@
...
 import { logger } from '../common/logger';
-import { getCommandLineArgs } from '../common/utils';
 import { config, IConfig } from './config-handler';
 
 // Set default flags
 logger.info(`chrome-flags: Setting mandatory chrome flags`, { flag: { 'ssl-version-fallback-min': 'tls1.2' } });
 app.commandLine.appendSwitch('ssl-version-fallback-min', 'tls1.2');
 
+// Special args that need to be excluded as part of the chrome command line switch
+const specialArgs = ['--url', '--multiInstance', '--userDataPath=', 'symphony://', '--inspect-brk', '--inspect'];
```
Comment (VishwasShashidhar) [inline src/app/chrome-flags.ts]: we also need to add `--logPath=` to this array.

File: src/renderer/ssf-api.ts [hunk 13]
```diff
@@ -170,12 +172,14 @@ export class SSFApi {
...
         const appName = remote.app.getName();
         const appVer = remote.app.getVersion();
+        const cpuArch = os.arch() || '';
 
         return Promise.resolve({
             containerIdentifier: appName,
             containerVer: appVer,
             buildNumber,
             apiVer: '2.0.0',
+            cpuArch,
```
Comment (VishwasShashidhar) [inline src/renderer/ssf-api.ts]: can we also add available memory information to this data?

**Comment by VishwasShashidhar** [comment]
@keerthi16 If everything is good, can you approve this PR?

**Pull request merged**